CSOAR-5032 - Documentation of GKE (Google Kubernetes Engine) (#6808)
SumoLogic/sumologic-documentation: docs/platform-services/automation-service/app-central/integrations/google-kubernetes-engine.md

* **List Cluster Role Bindings** (*Enrichment*) - Lists all cluster role bindings in a GKE cluster.
* **List Clusters** (*Enrichment*) - Lists all GKE clusters in the specified project and region.
* **List Workloads** (*Enrichment*) - Lists all workloads running in a GKE cluster.
* **Grant Cluster Role Binding** (*Containment*) - Creates a cluster role binding in a GKE cluster.
* **Revoke Cluster Role Binding** (*Containment*) - Deletes a cluster role binding from a GKE cluster.
* **Pod Security Monitor Daemon** (*Daemon*) - Monitors pod security events and alerts on policy violations in a GKE cluster.
* **Scale Down Deployment** (*Containment*) - Scales down the replica count of a specified deployment in a GKE cluster.

## Google Kubernetes Engine configuration

The Google Kubernetes Engine integration supports two types of authentication:
- **Service Account**
- **WIF (Workload Identity Federation)**

We recommend using WIF since it is more secure and easier to manage. For more information, see [Workload Identity Federation](https://cloud.google.com/iam/docs/workload-identity-federation).

## Required AWS details from Sumo Logic

> To configure the Google Kubernetes Engine integration using WIF authentication, you need the following AWS details from Sumo Logic. These details are essential for setting up the Workload Identity Federation (WIF) credentials in Google Workspace:

Reviewer: ```suggestion
To configure the Google Kubernetes Engine integration using WIF authentication, you need the following AWS details from Sumo Logic. These details are essential for setting up the Workload Identity Federation (WIF) credentials in Google Cloud Platform (GCP):
```

Author: Done

* Deployment name is the unique name of your Sumo Logic [deployment](/docs/api/about-apis/getting-started/#documentation), for example, `dub`, `fra`, etc. 
* Sumo Logic AWS account ID: `926226587429`
* Sumo Logic AWS role: `<deployment_name>-csoar-automation-gcpgke`
* Sumo Logic AWS Lambda function: `<deployment_name>-csoar-automation-gcpgke`
* Full ARN: `arn:aws:sts::926226587429:assumed-role/<deployment_name>-csoar-automation-gcpgke/<deployment_name>-csoar-automation-gcpgke`

### Workload Identity Federation (WIF) authentication

> Follow the steps below to [create WIF credentials](https://cloud.google.com/iam/docs/workload-identity-federation) in Google Workspace, which are required to configure the Google Kubernetes Engine integration:

Reviewer: ```suggestion
Follow the steps below to [create WIF credentials](https://cloud.google.com/iam/docs/workload-identity-federation) in GCP, which are required to configure the Google Kubernetes Engine integration:
```

Author: Done

15. Go to the **IAM & Admin** > **Workload Identity Federation** and select the pool created in step 9. 
16. Click **Grant Access** > **Grant access using service account impersonation**. 
17. Select the service account created in the previous step. Set the principal type as `aws_role` and the ARN as `arn:aws:sts::{SumoAWSAccountID}:assumed-role/{SumoAWSRole}` and then click **SAVE**. <br/><img src={useBaseUrl('/img/platform-services/automation-service/app-central/integrations/google-chat/google-chat-10.png')} style={{border:'1px solid gray'}} alt="Provide ARN" width="800"/>
18. Navigate to **Grant Access** > **Grant access using service account impersonation**. Select the service account created in step 6. Select the principal as `aws_role` and provide the ARN as `arn:aws:sts::{SumoAWSAccountID}:assumed-role/{SumoAWSRole}/{SumoAWSLambdaFunction}`. Click **SAVE**. 
19. Download the WIF `conf.json` file. Ensure you save it in a safe place. Use the JSON content to configure the Google Kubernetes Engine integration to use WIF authentication in Automation Service and Cloud SOAR. 

### Service Account authentication

> To [create service account credentials](https://developers.google.com/workspace/guides/create-credentials) in Google Workspace, needed to configure the Google Kubernetes Engine integration, follow these steps:

Reviewer: ```suggestion
To [create service account credentials](https://developers.google.com/workspace/guides/create-credentials) in GCP, needed to configure the Google Kubernetes Engine integration, follow these steps:
```

Author: Done
